csFastFloat/FastDoubleParser.cs

```diff
@@ -52,52 +67,296 @@ public  static double FastPath(ParsedNumberString pns)
+    public static unsafe double ParseDouble(char* first, char* last, NumberStyles styles = NumberStyles.Float, char decimal_separator = '.')
+      => ParseDouble(first, last, out int _, styles, decimal_separator);
+
+    /// <summary>
+    /// Parses double from a UTF-8 encoded readonly span of bytes  in the given number style
+    /// </summary>
+    /// <param name="s">input as a readonly span of bytes</param>
+    /// <param name="styles">allowed styles for the input string</param>
+    /// <param name="decimal_separator">decimal separator to be used</param>
+    /// <returns>parsed double value </returns>
+    public static unsafe double ParseDouble(ReadOnlySpan<byte> s, NumberStyles styles = NumberStyles.Float, byte decimal_separator = (byte)'.')
+      => ParseDouble(s, out int _, styles, decimal_separator);
+
+    /// <summary>
+    /// Parses double from a UTF-8 encoded readonly span of bytes in the given number style counting number of characters consummed
+    /// </summary>
+    /// <param name="s">input as a readonly span of bytes</param>
+    /// <param name="characters_consumed">number of character consummed while parsing</param>
+    /// <param name="styles">allowed styles for the input string</param>
+    /// <param name="decimal_separator">decimal separator to be used</param>
+    /// <returns>parsed double value </returns>
+    public static unsafe double ParseDouble(ReadOnlySpan<byte> s, out int characters_consumed, NumberStyles styles = NumberStyles.Float, byte decimal_separator = (byte)'.')
     {
-      fixed (char* pStart = s)
+      fixed (byte* pStart = s)
       {
-        return ParseNumber(pStart, pStart + (uint)s.Length, out characters_consumed, expectedFormat, decimal_separator);
+        return ParseDouble(pStart, pStart + s.Length, out characters_consumed, styles, decimal_separator);
       }
     }
 
-    unsafe static public double ParseDouble(char* first, char* last, NumberStyles expectedFormat = NumberStyles.Float, char decimal_separator = '.')
-      => ParseNumber(first, last, out int _, expectedFormat, decimal_separator);
+
+    /// <summary>
+    /// Parses double from a UTF-16 encoded input in the given number style counting number of characters consummed
+    /// </summary>
+    /// <param name="first">char pointer to the begining of the input</param>
+    /// <param name="last">char pointer to the end of the input</param>
+    /// <param name="characters_consumed">number of character consummed while parsing</param>
+    /// <param name="styles">allowed styles for the input string</param>
+    /// <param name="decimal_separator">decimal separator to be used</param>
+    /// <returns>parsed double value </returns>
+    public static unsafe double ParseDouble(char* first, char* last, out int characters_consumed, NumberStyles styles = NumberStyles.Float, char decimal_separator = '.')
+    {
+      if (!TryParseDouble(first, last, out characters_consumed, out double result, styles, decimal_separator))
```

Comment:
Maybe swap the condition? The branch predictor (when there's no history) will take the fall-through path. So 
```c#
if (TryParse(...))
{
  return result;
}

ThrowArgumentException();
throw null;
```
Note: the final `throw null` is just there to make the compiler happy, as otherwise there wouldn't be a return value from this code path.
Alternatively instead of `throw null` you can use `return default`. At codegen this will be dead code eliminated anyway.